repo/datastore.go

```diff
@@ -13,24 +14,31 @@ import (
 
 type Datastore interface {
 	Config() Config
-	Followers() Followers
-	Following() Following
-	OfflineMessages() OfflineMessages
-	Pointers() Pointers
-	Settings() Settings
-	Inventory() Inventory
-	Purchases() Purchases
-	Sales() Sales
-	Cases() Cases
-	Chat() Chat
-	Notifications() Notifications
-	Coupons() Coupons
-	TxMetadata() TxMetadata
-	ModeratedStores() ModeratedStores
+	Followers() FollowerStore
+	Following() FollowingStore
+	OfflineMessages() OfflineMessageStore
+	Pointers() PointerStore
+	Settings() ConfigurationStore
+	Inventory() InventoryStore
+	Purchases() PurchaseStore
+	Sales() SaleStore
+	Cases() CaseStore
+	Chat() ChatStore
+	Notifications() NotificationStore
+	Coupons() CouponStore
+	TxMetadata() TransactionMetadataStore
+	ModeratedStores() ModeratedStore
 	Ping() error
 	Close()
 }
 
+type Queryable interface {
+	BeginTransaction() (*sql.Tx, error)
+	PrepareQuery(string) (*sql.Stmt, error)
+	PrepareAndExecuteQuery(string, ...interface{}) (*sql.Rows, error)
+	ExecuteQuery(string, ...interface{}) (sql.Result, error)
+}
+
```

Comment:
This is how we will share future shared behavior across the Stores. Eventually, we'll migrate things like Put into here once we sort out the typecasting. We would approach this in steps and this is the foundation.